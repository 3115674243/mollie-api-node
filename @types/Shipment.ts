declare namespace Mollie {
  interface Shipment {
    lines: Array<OrderLine>;
    tracking?: Tracking;

    // Access token parameters
    testmode?: boolean;
  }

  interface OrderShipment extends Shipment {
    orderId?: string,
  }

  interface Tracking {
    carrier: string;
    code: string;
    url?: string;
  }
}
