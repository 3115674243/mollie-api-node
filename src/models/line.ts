import Model from '../model';

/**
 * The `line` model
 */
export default class Line extends Model implements Mollie.FullOrderLine {
  id?: string;
  type?: Mollie.OrderLineType;
  name: string;
  quantity: number;
  unitPrice: Mollie.Amount;
  discountAmount?: Mollie.Amount;
  totalAmount: Mollie.Amount;
  vatRate: string;
  vatAmount: Mollie.Amount;
  sku: string;
  imageUrl: string;
  productUrl: string;
  _links: Mollie.Links;

  constructor(props?: Partial<Mollie.FullOrderLine>) {
    super(props);

    const defaults: Mollie.FullOrderLine = {
      resource: null,
      id: null,
      name: null,
      quantity: null,
      unitPrice: null,
      discountAmount: null,
      totalAmount: null,
      vatRate: null,
      vatAmount: null,
      sku: null,
      imageUrl: null,
      productUrl: null,
      _links: {
        order: null,
        settlement: null,
      },
    };

    Object.assign(this, defaults, props);
  }
}
