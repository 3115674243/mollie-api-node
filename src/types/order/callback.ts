import { IOrder } from '../order';
import { IList } from '../list';

/**
 * @deprecated since 2.2.0 - All callbacks will be removed in a future version
 */
export type CreateCallback = (error: any, order?: IOrder) => void;
/**
 * @deprecated since 2.2.0 - All callbacks will be removed in a future version
 */
export type GetCallback = (error: any, order?: IOrder) => void;
/**
 * @deprecated since 2.2.0 - All callbacks will be removed in a future version
 */
export type ListCallback = (error: any, orders?: IList<IOrder>) => void;
/**
 * @deprecated since 2.2.0 - All callbacks will be removed in a future version
 */
export type UpdateCallback = (error: any, order?: IOrder) => void;
/**
 * @deprecated since 2.2.0 - All callbacks will be removed in a future version
 */
export type CancelCallback = (error: any, order?: IOrder) => void;
