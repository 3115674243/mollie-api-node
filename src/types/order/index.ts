import { ApiMode, IAddress, IAmount, ILinks } from '../global';
import { IOrderLine } from './line';

/**
 * Order Response object.
 *
 * @see https://docs.mollie.com/reference/v2/orders-api/get-order
 */
export interface IOrder {
  resource: string;
  id: string;
  profileId: string;
  method: string | null;
  mode: ApiMode;
  amount: IAmount;
  amountCaptured: IAmount | null;
  amountRefunded: IAmount | null;
  status: OrderStatus;
  isCancelable: boolean;
  billingAddress: IOrderAddress;
  consumerDateOfBirth?: string;
  orderNumber: string;
  shippingAddress: IOrderAddress;
  locale: string;
  metadata: any;
  redirectUrl: string | null;
  lines: Array<Partial<IOrderLine>>;
  webhookUrl?: string;
  createdAt: string;
  expiresAt?: string;
  expiredAt?: string;
  paidAt?: string;
  authorizedAt?: string;
  canceledAt?: string;
  completedAt?: string;
  _links: ILinks;
}

export type OrderStatus =
  | 'created'
  | 'paid'
  | 'authorized'
  | 'canceled'
  | 'shipping'
  | 'completed'
  | 'expired';

export interface IOrderAddress extends IAddress {
  title?: string;
  givenName: string;
  familyName: string;
  email: string;
  phone?: string;
}

export enum OrderEmbed {
  payments = 'payments',
}
