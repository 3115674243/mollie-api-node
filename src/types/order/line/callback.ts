import { IOrder } from '@mollie-types/order';

/**
 * @deprecated since 2.2.0 - All callbacks will be removed in a future version
 */
export type UpdateCallback = (error: any, order?: IOrder) => void;
/**
 * @deprecated since 2.2.0 - All callbacks will be removed in a future version
 */
export type CancelCallback = (error: any, success?: boolean) => void;
