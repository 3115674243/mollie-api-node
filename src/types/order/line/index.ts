import { IAmount, ILinks, IUrl } from '../../global';
import { OrderStatus } from '../../order';

/**
 * OrderLine Response object.
 */
export interface IOrderLine {
  resource: string;
  id: string;
  orderId?: string;
  type: OrderLineType;
  name: string;
  status: OrderStatus;
  isCancelable: boolean;
  quantity: number;
  quantityShipped: number;
  amountShipped: IAmount;
  quantityRefunded: number;
  amountRefunded: IAmount;
  quantityCanceled: number;
  amountCanceled: IAmount;
  shippableQuantity: number;
  refundableQuantity: number;
  cancelableQuantity: number;
  unitPrice: IAmount;
  discountAmount?: IAmount;
  totalAmount: IAmount;
  vatRate: string;
  vatAmount: IAmount;
  sku?: string;
  createdAt: string;
  _links: IOrderLineLinks;
  imageUrl: string;
  productUrl: string;
  metadata: any;
}

export interface IOrderLineLinks {
  productUrl: IUrl;
  imageUrl: IUrl;
}

export interface IRequestOrderLine {
  id: string;
  quantity?: number;
  amount?: IAmount;
}

export enum OrderLineType {
  physical = 'physical',
  discount = 'discount',
  digital = 'digital',
  shipping_fee = 'shipping_fee',
  store_credit = 'store_credit',
  gift_card = 'gift_card',
  surcharge = 'surcharge',
}
