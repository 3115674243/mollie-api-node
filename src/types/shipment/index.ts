import { ILinks, IUrl } from '../global';
import { IOrderLine } from '../order/line';

/**
 * Shipment Response object.
 *
 * @param resource - Indicates the response contains a shipment object. Will always contain shipment for this endpoint.
 * @param id - The shipment’s unique identifier, for example `shp_3wmsgCJN4U`.
 * @param orderId - The order this shipment was created on, for example `ord_8wmqcHMN4U`.
 * @param createdAt - The shipment’s date and time of creation, in ISO 8601 format.
 * @param tracking - @see {@link IShipmentTracking}
 * @param lines - An array of {@link IOrderLine order line objects} as described in {@link https://docs.mollie.com/reference/v2/shipments-api/get-shipment Get order}.
 *                The lines will show the quantity, discountAmount, vatAmount and totalAmount shipped in this shipment.
 *                If the line was partially shipped, these values will be different from the values in response from the
 *                Get order API.
 * @param _links - An object with several URL objects relevant to the shipment.
 *
 * @see https://docs.mollie.com/reference/v2/shipments-api/get-shipment
 */
export interface IShipment {
  resource: string;
  id: string;
  orderId: string;
  createdAt: string;
  tracking: IShipmentTracking | null;
  lines: Array<IOrderLine>;
  _links: IShipmentLinks;

  // Access token parameters
  testmode?: boolean;
}

/**
 * Shipments _links object
 *
 * @param order - The resource URL of the order this shipment was created for.
 */
export interface IShipmentLinks extends ILinks {
  order: IUrl;
}

/**
 * An object containing shipment tracking details.
 * Will be omitted when no tracking details are available.
 *
 * @param carrier - The name of the postal carrier.
 * @param code - The track and trace code for the shipment.
 * @param url - The URL where your customer can track the shipment.
 */
export interface IShipmentTracking {
  carrier: string;
  code: string;
  url?: string;
}
