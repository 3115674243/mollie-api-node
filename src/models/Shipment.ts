import Model from '../model';
import { IShipment } from '../types/shipment';

/**
 * The `Shipment` model
 *
 * {@link IShipment}
 */
export default class Shipment extends Model implements IShipment {
  public static resourcePrefix = 'shp_';

  public resource = 'shipment';
  public id = null;
  public orderId = null;
  public createdAt = null;
  public tracking = null;
  public lines = null;
  public _links = {
    self: null,
    order: null,
    documentation: null,
  };

  // Access token parameters
  public testmode?: boolean;

  /**
   * Shipment constructor
   *
   * @public âœ“ This constructor is part of the public API
   */
  public constructor(props?: Partial<IShipment>) {
    super();

    Object.assign(this, props);
  }
}
