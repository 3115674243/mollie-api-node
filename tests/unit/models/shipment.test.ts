import Shipment from '@models/Shipment';
import { IShipment } from '@mollie-types/shipment';
import { OrderStatus } from '@mollie-types/order';
import { OrderLineType } from '@mollie-types/order/line';

describe('shipment model', () => {
  it('should instantiate with default values', () => {
    const shipment = new Shipment();

    expect(shipment.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const shipmentProps: IShipment = {
      resource: 'shipment',
      id: 'shp_3wmsgCJN4U',
      orderId: 'ord_kEn1PlbGa',
      createdAt: '2018-08-09T14:33:54+00:00',
      tracking: {
        carrier: 'PostNL',
        code: '3SKABA000000000',
        url: 'http://postnl.nl/tracktrace/?B=3SKABA000000000&P=1016EE&D=NL&T=C',
      },
      lines: [
        {
          resource: 'orderline',
          id: 'odl_dgtxyl',
          orderId: 'ord_pbjz8x',
          name: 'LEGO 42083 Bugatti Chiron',
          sku: '5702016116977',
          type: OrderLineType.physical,
          status: OrderStatus.created,
          metadata: {
            order_id: '1337',
            description: 'Bugatti Chiron',
          },
          isCancelable: false,
          quantity: 2,
          quantityShipped: 0,
          amountShipped: {
            value: '0.00',
            currency: 'EUR',
          },
          quantityRefunded: 0,
          amountRefunded: {
            value: '0.00',
            currency: 'EUR',
          },
          quantityCanceled: 0,
          amountCanceled: {
            value: '0.00',
            currency: 'EUR',
          },
          shippableQuantity: 0,
          refundableQuantity: 0,
          cancelableQuantity: 0,
          unitPrice: {
            value: '399.00',
            currency: 'EUR',
          },
          vatRate: '21.00',
          vatAmount: {
            value: '121.14',
            currency: 'EUR',
          },
          discountAmount: {
            value: '100.00',
            currency: 'EUR',
          },
          totalAmount: {
            value: '698.00',
            currency: 'EUR',
          },
          createdAt: '2018-08-02T09:29:56+00:00',
          _links: {
            productUrl: {
              href: 'https://shop.lego.com/nl-NL/Bugatti-Chiron-42083',
              type: 'text/html',
            },
            imageUrl: {
              href: 'https://sh-s7-live-s.legocdn.com/is/image//LEGO/42083_alt1?$main$',
              type: 'text/html',
            },
          },
        },
        {
          resource: 'orderline',
          id: 'odl_jp31jz',
          orderId: 'ord_pbjz8x',
          name: 'LEGO 42056 Porsche 911 GT3 RS',
          sku: '5702015594028',
          type: OrderLineType.physical,
          status: OrderStatus.created,
          metadata: null,
          isCancelable: false,
          quantity: 1,
          quantityShipped: 0,
          amountShipped: {
            value: '0.00',
            currency: 'EUR',
          },
          quantityRefunded: 0,
          amountRefunded: {
            value: '0.00',
            currency: 'EUR',
          },
          quantityCanceled: 0,
          amountCanceled: {
            value: '0.00',
            currency: 'EUR',
          },
          shippableQuantity: 0,
          refundableQuantity: 0,
          cancelableQuantity: 0,
          unitPrice: {
            value: '329.99',
            currency: 'EUR',
          },
          vatRate: '21.00',
          vatAmount: {
            value: '57.27',
            currency: 'EUR',
          },
          totalAmount: {
            value: '329.99',
            currency: 'EUR',
          },
          createdAt: '2018-08-02T09:29:56+00:00',
          _links: {
            productUrl: {
              href: 'https://shop.lego.com/nl-NL/Porsche-911-GT3-RS-42056',
              type: 'text/html',
            },
            imageUrl: {
              href: 'https://sh-s7-live-s.legocdn.com/is/image/LEGO/42056?$PDPDefault$',
              type: 'text/html',
            },
          },
        },
      ],
      _links: {
        self: {
          href: 'https://api.mollie.com/v2/order/ord_kEn1PlbGa/shipments/shp_3wmsgCJN4U',
          type: 'application/hal+json',
        },
        order: {
          href: 'https://api.mollie.com/v2/orders/ord_kEn1PlbGa',
          type: 'application/hal+json',
        },
        documentation: {
          href: 'https://docs.mollie.com/reference/v2/shipments-api/get-shipment',
          type: 'text/html',
        },
      },
    };

    const shipment = new Shipment(shipmentProps);

    expect(shipment.toPlainObject()).toMatchSnapshot();
  });
});
