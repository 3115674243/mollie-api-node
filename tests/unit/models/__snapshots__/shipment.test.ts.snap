// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shipment model should instantiate with default values 1`] = `
Object {
  "_links": Object {
    "documentation": null,
    "order": null,
    "self": null,
  },
  "createdAt": null,
  "id": null,
  "lines": null,
  "mode": undefined,
  "orderId": null,
  "resource": "shipment",
  "testmode": undefined,
  "tracking": null,
}
`;

exports[`shipment model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "documentation": Object {
      "href": "https://docs.mollie.com/reference/v2/shipments-api/get-shipment",
      "type": "text/html",
    },
    "order": Object {
      "href": "https://api.mollie.com/v2/orders/ord_kEn1PlbGa",
      "type": "application/hal+json",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/order/ord_kEn1PlbGa/shipments/shp_3wmsgCJN4U",
      "type": "application/hal+json",
    },
  },
  "createdAt": "2018-08-09T14:33:54+00:00",
  "id": "shp_3wmsgCJN4U",
  "lines": Array [
    Object {
      "_links": Object {
        "imageUrl": Object {
          "href": "https://sh-s7-live-s.legocdn.com/is/image//LEGO/42083_alt1?$main$",
          "type": "text/html",
        },
        "productUrl": Object {
          "href": "https://shop.lego.com/nl-NL/Bugatti-Chiron-42083",
          "type": "text/html",
        },
      },
      "amountCanceled": Object {
        "currency": "EUR",
        "value": "0.00",
      },
      "amountRefunded": Object {
        "currency": "EUR",
        "value": "0.00",
      },
      "amountShipped": Object {
        "currency": "EUR",
        "value": "0.00",
      },
      "cancelableQuantity": 0,
      "createdAt": "2018-08-02T09:29:56+00:00",
      "discountAmount": Object {
        "currency": "EUR",
        "value": "100.00",
      },
      "id": "odl_dgtxyl",
      "isCancelable": false,
      "metadata": Object {
        "description": "Bugatti Chiron",
        "order_id": "1337",
      },
      "name": "LEGO 42083 Bugatti Chiron",
      "orderId": "ord_pbjz8x",
      "quantity": 2,
      "quantityCanceled": 0,
      "quantityRefunded": 0,
      "quantityShipped": 0,
      "refundableQuantity": 0,
      "resource": "orderline",
      "shippableQuantity": 0,
      "sku": "5702016116977",
      "status": "created",
      "totalAmount": Object {
        "currency": "EUR",
        "value": "698.00",
      },
      "type": "physical",
      "unitPrice": Object {
        "currency": "EUR",
        "value": "399.00",
      },
      "vatAmount": Object {
        "currency": "EUR",
        "value": "121.14",
      },
      "vatRate": "21.00",
    },
    Object {
      "_links": Object {
        "imageUrl": Object {
          "href": "https://sh-s7-live-s.legocdn.com/is/image/LEGO/42056?$PDPDefault$",
          "type": "text/html",
        },
        "productUrl": Object {
          "href": "https://shop.lego.com/nl-NL/Porsche-911-GT3-RS-42056",
          "type": "text/html",
        },
      },
      "amountCanceled": Object {
        "currency": "EUR",
        "value": "0.00",
      },
      "amountRefunded": Object {
        "currency": "EUR",
        "value": "0.00",
      },
      "amountShipped": Object {
        "currency": "EUR",
        "value": "0.00",
      },
      "cancelableQuantity": 0,
      "createdAt": "2018-08-02T09:29:56+00:00",
      "id": "odl_jp31jz",
      "isCancelable": false,
      "metadata": null,
      "name": "LEGO 42056 Porsche 911 GT3 RS",
      "orderId": "ord_pbjz8x",
      "quantity": 1,
      "quantityCanceled": 0,
      "quantityRefunded": 0,
      "quantityShipped": 0,
      "refundableQuantity": 0,
      "resource": "orderline",
      "shippableQuantity": 0,
      "sku": "5702015594028",
      "status": "created",
      "totalAmount": Object {
        "currency": "EUR",
        "value": "329.99",
      },
      "type": "physical",
      "unitPrice": Object {
        "currency": "EUR",
        "value": "329.99",
      },
      "vatAmount": Object {
        "currency": "EUR",
        "value": "57.27",
      },
      "vatRate": "21.00",
    },
  ],
  "mode": undefined,
  "orderId": "ord_kEn1PlbGa",
  "resource": "shipment",
  "testmode": undefined,
  "tracking": Object {
    "carrier": "PostNL",
    "code": "3SKABA000000000",
    "url": "http://postnl.nl/tracktrace/?B=3SKABA000000000&P=1016EE&D=NL&T=C",
  },
}
`;
